pennylane/ops/functions/generator.py

```diff
@@ -78,11 +78,11 @@ def _generator_prefactor(gen, op):
       :class:`~.SparseHamiltonian`, or a tensor product
       of Pauli words.
     """
-    if isinstance(gen, (qml.Hermitian, qml.SparseHamiltonian)):
-        obs = gen
-        prefactor = 1.0
 
-    elif isinstance(gen, qml.operation.Observable):
+    obs = gen
+    prefactor = 1.0
+
+    if isinstance(gen, qml.operation.Observable):
```

Comment:
I don't think you need the else, because you defined 'obs' and 'prefactor' before

Reply:
I believe the variable `gen` has been converted to a Hamiltonian inside the `if Observable` statement (in line 87), and so setting `obs=gen` inside the `else` statement is not equivalent to what was done previously.